### DISCUSSION TITLE
Button reload when browser resize

### DISCUSSION THREAD

**Discussion by smartlightning** [rpldy/react-uploady#405] [Q&A]
When I upload an image and resize the screen the state in my custom button is reset and I never enter the  custom useBatchFinishListener hook. 
Anyone else having this issue?

**Comment by yoavniran** [comment]
@smartlightning it sounds like screen resize causes the Uploady component to re-render and create a new internal uploader due to one of the props you pass to it not have the same value. most likely a function or object/array being created on each re-render.

Its suggested to put to Uploady somewhere high enough in the render tree where it wont re-render frequently (or at all) and if it does, make sure you're passing same values. leveraging useCallback, useMemo as required.

You should examine your render tree and see whats causing this.
